consoles/virtio_terminal.pm

```diff
@@ -110,33 +108,38 @@ sub load_snapshot {
 
   open_socket();
 
-Opens a socket to the host end of the virtio_console. This need to be
-corresponding to the console which was created by qemu-backend.
+Opens a the read and write pipe based on C<$pipe_prefix>.
 
-Returns the file descriptor for the open socket, otherwise it dies.
+Returns the read and write file descriptors for the open sockets,
+otherwise it dies.
 
 =cut
 sub open_socket {
     my ($self) = @_;
-    my $fd;
-    my $port;
-    my $vars = \%bmwqemu::vars;
-    bmwqemu::log_call(console_num => $self->{console_num});
-
-    $port = 62600 + $self->{console_num} + 10 * ($vars->{WORKER_INSTANCE} // 0);
-    $fd   = IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => $port, Proto => 'tcp')
-      or die('Connection to virtio_terminal nr ' . $self->{console_num} . ' failed on port ' . $port);
+    bmwqemu::log_call(pipe_prefix => $self->{pipe_prefix});
+
+    sysopen(my $fd_w, $self->{pipe_prefix} . '.in', O_WRONLY)
+      or die "Can't open in pipe for writing $!";
+    sysopen(my $fd_r, $self->{pipe_prefix} . '.out', O_NONBLOCK | O_RDONLY)
+      or die "Can't open out pipe for reading $!";
+
+    my $newsize = get_var('VIRTIO_CONSOLE_PIPE_SZ', 1024 * 1024);
```

Comment:
You could read the max pipe size from the `/proc/sys/fs/pipe-max-size` and use that. On kernel's after 4.9 it might be set to that by default anyway. On older kernels or systems with unusual page sizes the limit might not be 1MB either. I don't know when a user would need to set a pipe size that is not the max for their system; they would need to have some huge max size for it to make sense.

You might already know this, but you may need to use `sysopen` and `sysread` on this file then parse the resulting string as an integer. The normal Perl functions might not work because these files don't implement `size` and stuff like that.

Reply:
Good idea, I will read out the default `$newsize` from that file.
Thx for the hints.

Reply:
I'm not sure if I get the point regarding `sysopen/sysread`. I just tried a 
```
perl -e 'use Mojo::File "path"; print path("/proc/sys/fs/pipe-max-size")->slurp();'
```
and it worked for me.